test/core/security/credentials_test.cc

```diff
@@ -2451,21 +2499,110 @@ TEST(CredentialsTest,
   HttpRequest::SetOverride(nullptr, nullptr, nullptr);
 }
 
+TEST(
+    CredentialsTest,
+    TestExternalAccountCredsSuccessWithServiceAccountImpersonationAndCustomTokenLifetime) {
+  ExecCtx exec_ctx;
+  Json credential_source = Json::FromString("");
+  auto service_account_impersonation =
+      JsonParse("{\"token_lifetime_seconds\":1800}");
```

Comment:
Could you also add tests where expected names contain typos to make sure code correctly handles such cases? Ie `"service_account_impersonationZ"` and `"token_lifetime_secondsZ"`.

Reply:
We are creating the "options" object manually and using it to create a "Test External Acc Credential". So there is no json to credential creation happening here to test the misspelled keys. Also this type of test for misspelled key has not been done for any other credential types.